### PR TITLE
feat(pricing,medusa,utils): added list + get endpoints for price lists

### PR DESCRIPTION
**Pull request by riqwan**
what:

- brings back price list GET endpoints to v2 endpoints

### CODE REVIEW

File: packages/medusa/src/api/routes/admin/price-lists/modules-queries/get-price-list.ts
Comment (sradevski) [inline packages/medusa/src/api/routes/admin/price-lists/modules-queries/get-price-list.ts:15]: Doesn't this work if it is `filters: { id }`?

File: packages/medusa/src/api-v2/admin/price-lists/query-config.ts
Comment (sradevski) [inline packages/medusa/src/api-v2/admin/price-lists/query-config.ts:40]: Does this match what we had in v1? If we can match it than that'd be great, if not then we'll have to live with the breaking changes :)

Comment (riqwan) [reply packages/medusa/src/api-v2/admin/price-lists/query-config.ts:40]: Yes, the v2 response matches v1 response. The transformation of v2 to v1 happens in the utils function. This is because the data model of pricing between v1 and v2 is vastly different. 

This variable and the ones beneath this is however wrong though. Its meant to showcase a list of attributes that should be present in the response. But in reality, this is a list of remote query fields to query through the price_list entry point. 

I'll look into it a bit more to see how we can clearly depict the 2.

File: packages/medusa/src/api/routes/admin/price-lists/modules-queries/get-price-list.ts
Comment (riqwan) [reply packages/medusa/src/api/routes/admin/price-lists/modules-queries/get-price-list.ts:15]: It will, this file will be deleted soon though 😄 its legacy stuff. Just waiting on one more endpoint before I nuke this.

File: packages/medusa/src/api-v2/admin/price-lists/query-config.ts
Comment (sradevski) [reply packages/medusa/src/api-v2/admin/price-lists/query-config.ts:40]: Ok cool, there is a lot going on here so I won't pretend I can judge what's a good approach, I was just wondering if before as well we expanded so much by default

File: packages/medusa/src/api-v2/admin/price-lists/query-config.ts
Comment (olivermrbl) [inline packages/medusa/src/api-v2/admin/price-lists/query-config.ts]: thought: Should we add a `fieldAlias` to the link definition, so we don't have to query this `variant_link` directly?

Something like:
```
{
      serviceName: Modules.PRICING,
      relationship: {
        serviceName: LINKS.ProductVariantPriceSet,
        primaryKey: "price_set_id",
        foreignKey: "id",
        alias: "variant_link",
      },
      fieldAlias: {
        variant: "variant_link.variant"
      },
    },
```

This would reduce the field to `price_set_money_amounts.price_set.variant`

Comment (olivermrbl) [reply packages/medusa/src/api-v2/admin/price-lists/query-config.ts]: Also, maybe we should review the UI to see if the variants on money amount are still required. It seems more appropriate to do a separate query in case we need them.

File: integration-tests/modules/__tests__/price-lists/admin/price-lists.spec.ts
Comment (olivermrbl) [inline integration-tests/modules/__tests__/price-lists/admin/price-lists.spec.ts]: todo: Should be updated to use the new `medusaIntegrationTestRunner`

File: packages/medusa/src/api-v2/admin/price-lists/query-config.ts
Comment (olivermrbl) [inline packages/medusa/src/api-v2/admin/price-lists/query-config.ts]: question: Actually, do we need all of this stuff to be default on the list endpoint? I can see why it could make sense on retrieval, but we don't need all prices and their variants to display the price list table, do we?. 

And if we do, I am almost leaning toward changing the UI rather than having this colossal query

File: packages/medusa/src/api-v2/admin/price-lists/utils/list-price-lists.ts
Comment (olivermrbl) [inline packages/medusa/src/api-v2/admin/price-lists/utils/list-price-lists.ts]: See comments on `query-config.ts`

**Review by olivermrbl — APPROVED** [review]
LGTM, but let's make sure to align on the query config before we merge

File: packages/medusa/src/api-v2/admin/price-lists/query-config.ts
Comment (riqwan) [reply packages/medusa/src/api-v2/admin/price-lists/query-config.ts]: We added it to make it compatible with the existing API response, back when enabling modules was still backwards compatible. 

We can definitely remove most of it and just keep the ID and query the variants API on the FE.

Comment (riqwan) [reply packages/medusa/src/api-v2/admin/price-lists/query-config.ts]: No, we don't. I've made some changes here that aligns with this, will ping you when I've pushed.

Comment (olivermrbl) [reply packages/medusa/src/api-v2/admin/price-lists/query-config.ts]: Nice, sounds good!

Comment (riqwan) [reply packages/medusa/src/api-v2/admin/price-lists/query-config.ts]: @olivermrbl take a look at the latest commit, this is the prettiest I could make for now. 😄

The `fields` and `expands` is not friendly to the remote query approach, since the api has to know how the public interface is mapped to both the modules.

File: packages/medusa/src/utils/omit-deep.ts
Comment (olivermrbl) [inline packages/medusa/src/utils/omit-deep.ts]: nit: Move to separate file `pick-deep.ts`

**Review by olivermrbl — COMMENTED** [review]
Think this looks better. We just need tests to pass, and I would be fine with merging 💪

**Pull request merged**
